syntax = "proto3";

service Server {
 rpc Register(RegisterRequest) returns (RegisterResponse) {}
 rpc UserMessage(UserMessageRequest) returns (UserMessageResponse) {}
}

message RegisterRequest {
 string username = 1;
 string hostname = 2;
 int32 port = 3;
 string uuid = 4;
}

message RegisterResponse {
  string username = 1;
  int32 session_id = 2;
  string role = 3;
}

message UserMessageRequest {
  string uuid = 1;
  int32 session_id = 2;
  string message = 3;
  bool secret = 4;
 }
 
 message UserMessageResponse {
 }
